cmd/abapEnvironmentCreateTag.go

```diff
@@ -0,0 +1,267 @@
+
+	if err = json.Unmarshal(bodyText, &abapResp); err != nil {
+		return err
+	}
+	if err = json.Unmarshal(*abapResp["d"], &createTagResponse); err != nil {
+		return err
+	}
+
+	con.URL = con.Host + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull(guid'" + createTagResponse.UUID + "')"
+	err = checkStatus(con, client, com)
+
+	if err == nil {
+		log.Entry().Info("Created tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	} else {
+		log.Entry().Error("NOT created: Tag " + requestBodyStruct.Tag + " for repository " + requestBodyStruct.RepositoryName + " with commitID " + requestBodyStruct.CommitID)
+	}
+
+	return err
+}
+
+func checkStatus(con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+	status := "R"
+	pollIntervall := com.GetPollIntervall()
+	count := 0
+	for {
+		count += 1
+		entity, _, err := abaputils.GetStatus("Could not create Tag", con, client)
+		if err != nil {
+			return err
+		}
+		status = entity.Status
+		if status != "R" {
+			if status == "E" {
+				err = errors.New("Could not create Tag")
+			}
+			return err
+		}
+		if count >= 200 {
+			return errors.New("Could not create Tag (Timeout)")
+		}
+		time.Sleep(pollIntervall)
+	}
+}
+
+func prepareBacklog(config *abapEnvironmentCreateTagOptions) (backlog []CreateTagBacklog, err error) {
+
+	if config.Repositories != "" && config.RepositoryName != "" {
+		return nil, errors.New("Configuring the parameter repositories and the parameter repositoryName at the same time is not allowed")
+	}
+
+	descriptor := abaputils.AddonDescriptor{}
+	repos := []abaputils.Repository{}
+	if config.Repositories != "" {
+		descriptor, err = abaputils.ReadAddonDescriptor(config.Repositories)
+		if err != nil {
+			return nil, err
+		}
+		repos = descriptor.Repositories
+	}
+
+	for _, repo := range repos {
+
+		backlogInstance := CreateTagBacklog{repositoryName: repo.Name, commitID: repo.CommitID}
+		if config.GenerateTagForAddonComponentVersion && repo.VersionYAML != "" {
+			tag := Tag{tagName: "v" + repo.VersionYAML, tagDescription: "Generated by the ABAP Environment Pipeline"}
+			backlogInstance.tags = append(backlogInstance.tags, tag)
+		}
+		backlog = append(backlog, backlogInstance)
+	}
+
+	if config.GenerateTagForAddonProductVersion {
+		if descriptor.AddonProduct != "" && descriptor.AddonVersionYAML != "" {
+			addonProductDash := strings.Replace(descriptor.AddonProduct, "/", "-", 2)
+			backlog = addTagToList(backlog, addonProductDash+"-"+descriptor.AddonVersionYAML, "Generated by the ABAP Environment Pipeline")
+		} else {
+			log.Entry().WithField("generateTagForAddonProductVersion", config.GenerateTagForAddonProductVersion).WithField("AddonProduct", descriptor.AddonProduct).WithField("AddonVersion", descriptor.AddonVersionYAML).Infof("Not all required values are provided to create an addon product version tag")
+		}
+	}
```

Comment:
```suggestion
		for _, repo := range repos {

			backlogInstance := CreateTagBacklog{repositoryName: repo.Name, commitID: repo.CommitID}
			if config.GenerateTagForAddonComponentVersion && repo.VersionYAML != "" {
				tag := Tag{tagName: "v" + repo.VersionYAML, tagDescription: "Generated by the ABAP Environment Pipeline"}
				backlogInstance.tags = append(backlogInstance.tags, tag)
			}
			backlog = append(backlog, backlogInstance)
		}

		if config.GenerateTagForAddonProductVersion {
			if descriptor.AddonProduct != "" && descriptor.AddonVersionYAML != "" {
				addonProductDash := strings.Replace(descriptor.AddonProduct, "/", "-", 2)
				backlog = addTagToList(backlog, addonProductDash+"-"+descriptor.AddonVersionYAML, "Generated by the ABAP Environment Pipeline")
			} else {
				log.Entry().WithField("generateTagForAddonProductVersion", config.GenerateTagForAddonProductVersion).WithField("AddonProduct", descriptor.AddonProduct).WithField("AddonVersion", descriptor.AddonVersionYAML).Infof("Not all required values are provided to create an addon product version tag")
			}
		}
	}	
```
I would move anything which is only executed if the if = true into the if